File: 01-backend/src/main/java/ee/insa/crmapp/model/Firm.java
package ee.insa.crmapp.model;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "firm")
public class Firm {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    private String name;
    private String code;
    private String address;
    private String phone;
    private String fax;
    private String email;
    @OneToOne
    private Bank bank;
    private String bankAccount;
    private String comment;

    public Firm(String name, String code, String address, String phone, String fax, String email, Bank bank, String bankAccount, String comment) {
        this.name = name;
        this.code = code;
        this.address = address;
        this.phone = phone;
        this.fax = fax;
        this.email = email;
        this.bank = bank;
        this.bankAccount = bankAccount;
        this.comment = comment;
    }
}

File: 02-frontend/src/app/model/firm.ts
import {Bank} from "./bank";

export class Firm {
  public id: number;
  public name: string;
  public code: string;
  public address: string;
  public phone: string;
  public fax: string;
  public email: string;
  public bank: Bank;
  public bankAccount: string;
  public comment: string;

  constructor() {
    this.id = 0;
    this.name = '';
    this.code = '';
    this.address = '';
    this.phone = '';
    this.fax = '';
    this.email = '';
    this.bank = null;
    this.bankAccount = '';
    this.comment = '';
  }
}
